### DISCUSSION TITLE
Add basic auth for REST API

### DISCUSSION THREAD

**Discussion by Vlador15** [netbox-community/netbox#15576] [Help Wanted!] [answered]
Help me solve the problem. I don't understand how to log in via the REST API and get a token (the methods don't work without it). Basic auth does not work (headers.set('Authorization', `Basic ${btoa('admin:admin')}`);. Are there any options to create a token by username-password?

**Comment by markkuleinio** [comment]
https://docs.netbox.dev/en/stable/integrations/rest-api/#initial-token-provisioning

**Reply by Vlador15** [reply]
Thank you, I saw this headline, but I ran into a problem getting a token. We want to create a new user through REST and get a token for him. Using the /provisions method and passing username/password, I get an error in response: {"detail":"Authentication credentials were not provided."}. It is not possible to create a token for new users. Therefore, I thought that there are other ways to generate a token, for example by username:password through basic authentication. Tell me, have you ever encountered anything like this?

**Reply by markkuleinio** [reply, marked as answer]
Well, works here as documented:

```
$ curl -X POST \
-H "Content-Type: application/json" \
-H "Accept: application/json; indent=4" \
http://netbox-test.lein.io/api/users/tokens/provision/ \
--data '{
    "username": "just_created_user",
    "password": "test"
}'
( This is the response: )
{
    "id": 1,
    "url": "http://netbox-test.lein.io/api/users/tokens/1/",
    "display": "185225b3956b5a208321f8203dc36b1a8e05de0b",
    "user": {
        "id": 2,
        "url": "http://netbox-test.lein.io/api/users/users/2/",
        "display": "just_created_user",
        "username": "just_created_user"
    },
    "created": "2024-03-29T19:23:16.307642+02:00",
    "expires": null,
    "last_used": null,
    "key": "185225b3956b5a208321f8203dc36b1a8e05de0b",
    "write_enabled": true,
    "description": "",
    "allowed_ips": []
}
```

**Comment by candlerb** [comment]
> Basic auth does not work

That's correct, it doesn't use Basic auth headers. The header you need to provide is documented here:
https://docs.netbox.dev/en/stable/integrations/rest-api/#authenticating-to-the-api

**Reply by Vlador15** [reply]
Thank you, I saw this headline, but I ran into a problem getting a token. We want to create a new user through REST and get a token for him. Using the /provisions method and passing username/password, I get an error in response: {"detail":"Authentication credentials were not provided."}. It is not possible to create a token for new users. Therefore, I thought that there are other ways to generate a token, for example by username:password through basic authentication. Tell me, have you ever encountered anything like this?
